### PR TITLE
Remove shared_shorthand2xccdf.xslt

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- This PR removes the 'shorthand2xccdf' XSLT transformation.
- The commits go by integrating each XSLT transformation into the shorthand build process.
- A few typos and misplaced references were fixed.
  - Note that reference that were missing URIs were not fixed as part of this PR.

#### Rationale:

- Brings the `shorthand.xml` closer to being an actual XCCDF file, and possibly a XCCDF-1.2.

#### Next steps
- I believe the next step is to stop using the OpenSCAP `xccdf_1.1_to_1.2.xsl`.
  - The biggest impact would be that the shorthand would have the ID with the reverse DNS prefix, but a lot of our tooling and tests relies on the shorthand ID.

### CODE REVIEW

**Comment by yuumasato** [comment]
@dodys Hi, 1205e8fe6815492632f3be15adb3db5afded138c may affect the SCE checks.

If I understand correctly the SCE checks may be evaluated together with an OVAL check, but I'm not sure how it is working, as the one SCE check in upstream doesn't define a `complex-check`.

**Comment by jan-cerny** [comment]
I haven't seen this yet but @matejak and @Honny1 have an idea to replace the code generating XCCDF XML elements by a Jinja template.

**Comment by jan-cerny** [comment]
>  Brings the shorthand.xml closer to being an actual XCCDF file, and possibly a XCCDF-1.2.

The shorthand.xml now looks really like actual XCCDF file. What is missing there to make it an actual valid XCCDF? It seems that it's very close. Maybe we can rename it?

**Comment by JAORMX** [comment]
huh... this doesn't look right...
```
     helpers.go:807: E2E-FAILURE: The expected result for the api_server_audit_log_maxbackup rule didn't match. Expected 'PASS', Got 'FAIL' 
```
Not sure if this failure was introduced in this PR. But something definitely is changing the behavior of these rules.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_sudoers/rule.yml [hunk 1]
```diff
@@ -27,8 +27,8 @@ identifiers:
     cce@rhel9: CCE-90176-9
 
 references:
-    disa: CCI-000169
-    srg: SRG-OS-000004-GPOS-00004,SRG-OS-000037-GPOS-00015,SRG-OS-000042-GPOS-00020,SRG-OS-000062-GPOS-00031,SRG-OS-000304-GPOS-00121,SRG-OS-000392-GPOS-00172,SRG-OS-000462-GPOS-00206,SRG-OS-000470-GPOS-00214,SRG-OS-000471-GPOS-00215,SRG-OS-000239-GPOS-00089,SRG-OS-000240-GPOS-00090,SRG-OS-000241-GPOS-00091,SRG-OS-000303-GPOS-00120,SRG-OS-000304-GPOS-00121,CCI-002884,SRG-OS-000466-GPOS-00210,SRG-OS-000476-GPOS-00221
```
Comment (jan-cerny) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_sudoers/rule.yml:31]: Amazing!

**Review by jan-cerny — COMMENTED** [review]
I like it very much! Great job!

File: products/firefox/transforms/shorthand2xccdf.xslt [hunk 2]
```diff
@@ -1,9 +0,0 @@
-<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
-
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply products/firefox/transforms/shorthand2xccdf.xslt:2]: Should be fixed by 2b4f8e8

File: ssg/build_yaml.py [hunk 3]
```diff
@@ -877,7 +889,18 @@ def to_xml_element(self):
...
         version = ET.SubElement(root, 'version')
         version.text = self.version
-        ET.SubElement(root, "metadata")
+        version.set('update', SSG_BENCHMARK_LATEST_URI)
+        metadata = ET.SubElement(root, "metadata")
+        publisher = ET.SubElement(metadata, "{%s}publisher" % dc_namespace)
+        publisher.text = SSG_PROJECT_NAME
+        creator = ET.SubElement(metadata, "{%s}creator" % dc_namespace)
+        creator.text = SSG_PROJECT_NAME
+        contrib_tree = parse_file("../../Contributors.xml")
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply ssg/build_yaml.py]: Handle by 045313d, I hope it seems better.

File: ssg/build_yaml.py [hunk 4]
```diff
@@ -1453,49 +1483,54 @@ def validate_prodtype(self, yaml_file):
...
         add_sub_element(rule, 'title', self.title)
         add_sub_element(rule, 'description', self.description)
-        add_sub_element(rule, 'rationale', self.rationale)
+        add_warning_elements(rule, self.warnings)
+        if env_yaml:
+            ref_dict = env_yaml['reference_uris']
+        else:
+            ref_dict = SSG_REF_URIS
+        for ref_type, ref_vals in self.references.items():
+            for ref_val in ref_vals.split(","):
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply ssg/build_yaml.py]: Extracted into a separate function in cfa2263

**Comment by yuumasato** [comment]
> huh... this doesn't look right...
> 
> ```
>      helpers.go:807: E2E-FAILURE: The expected result for the api_server_audit_log_maxbackup rule didn't match. Expected 'PASS', Got 'FAIL' 
> ```
> 
> Not sure if this failure was introduced in this PR. But something definitely is changing the behavior of these rules.

@JAORMX From what I checked there is not relevant change in OCP4 or RHCOS4 DS that could cause rule behavior to change.
I rebased to latest master to see if these issues go away.

**Review by jan-cerny — APPROVED** [review]
I have compared the built artifacts with and without your patches and I haven't found any undesired changes. LGTM.

**Comment by matejak** [comment]
Awesome, the added/removed stat is really impressive, even more so given that the result is actually shorthand closer to xccdf.

**Pull request merged**